src/components/CalendarButtonTwoStep/CalendarButtonTwoStep.js

```diff
@@ -0,0 +1,163 @@
+  state = { startDate: null, endDate: null, focusedInput: null };
+  buttonRef = null;
+
+  _onDatesChange = ({ startDate, endDate }) => {
+    this.setState({ startDate, endDate });
+
+    if (!endDate) {
+      this._onFocusChange(END_DATE);
+    } else {
+      this._onFocusChange(START_DATE);
+    }
+  };
+
+  _onFocusChange = focusedInput => this.setState({ focusedInput });
+
+  _onClear = () =>
+    this.setState({ startDate: null, endDate: null, focusedInput: null });
+
+  _onButtonClick = () =>
+    this.setState(({ focusedInput }) => ({
+      focusedInput: focusedInput !== null ? null : START_DATE
+    }));
+
+  _currentSelection = () => {
+    const { startDate, endDate } = this.state;
+
+    if (!startDate && !endDate) {
+      return 'Dates';
+    }
+
+    return `${this._toDate(startDate)} - ${this._toDate(endDate)}`;
+  };
+
+  _toDate = date => (date ? date.format('MMM DD') : '');
+
+  _onButtonRef = ref => {
+    this.buttonRef = ref;
+  };
+
+  _onOutsideClick = ({ target }) => {
+    if (this.buttonRef) {
+      // TODO: May be implement forwardRef after we upgrade to 16.3 or elementRef
+      // eslint-disable-next-line react/no-find-dom-node
+      const buttonDomNode = findDOMNode(this.buttonRef);
+
+      if (!buttonDomNode.contains(target)) {
+        this._onFocusChange(null);
+      }
+    }
+  };
+
+  _onCommit = () => {
+    const { startDate, endDate } = this.state;
+
+    if (!startDate || !endDate) {
+      return;
+    }
+
+    this.props.onDatesRangeChange({ startDate, endDate });
+    this._onFocusChange(null);
+  };
+
+  render() {
+    const { focusedInput, startDate, endDate } = this.state;
+    const isOpen = focusedInput !== null;
+    const isFilled = !!(startDate && endDate);
+
+    return (
+      <CalendarButtonWrap>
+        <Button
+          primary={isOpen || isFilled}
+          ref={this._onButtonRef}
+          onClick={this._onButtonClick}
+        >
+          {this._currentSelection()}
+        </Button>
+        {isOpen && (
+          <CalendarWrap>
+            <RangePickerController
+              startDate={startDate}
+              endDate={endDate}
+              onDatesChange={this._onDatesChange}
+              focusedInput={focusedInput}
+              // onFocusChange={this._onFocusChange}
+              onOutsideClick={this._onOutsideClick}
+              numberOfMonths={2}
+              calendarInfoPosition="bottom"
+              renderCalendarInfo={() => (
+                <CalendarInfo>
+                  <InfoButton onClick={this._onClear}>Clear</InfoButton>
+                  <InfoButton primary onClick={this._onCommit}>
+                    Apply
+                  </InfoButton>
+                </CalendarInfo>
+              )}
+            />
+          </CalendarWrap>
+        )}
+      </CalendarButtonWrap>
+    );
+  }
+}
+
+const calendarInfoBase = () => css`
+  text-align: right;
+  margin: 0 23px 0; /* based on react dates */
```

Comment:
Hm, did @melmdesign not provide the Circuit spacings for this?

Reply:
The purpose of this is to align it with `react-dates` internal paddings. I've avoided touching them too much, because they have variables and calculations in the code.